code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1414,3 +1415,356 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
+
+
+/obj/structure/centrifuge
+	name = "suspicious toilet"
+	desc = "This toilet is a cleverly disguised improvised centrifuge."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "toilet11"
+	density = 0
+	anchored = 1
+	var/list/cans = new/list() //These are the empty containers.
+	var/obj/item/weapon/reagent_containers/beaker = null // This is the active container
+
+/obj/structure/centrifuge/examine(mob/user)
+	..()
+	user << "<span class='info'>It contains [cans.len] empty containers[beaker ? " and an active container!" : "."]</span>"
+
+/obj/structure/centrifuge/attackby(obj/item/weapon/reagent_containers/W as obj, mob/user as mob)
+	if(iscrowbar(W))
+		var/obj/structure/toilet/T = new /obj/structure/toilet(src.loc)
+		T.open = 1
+		T.cistern = 1
+		T.dir = src.dir
+		T.update_icon()
+		new /obj/item/stack/rods(get_turf(src), 2)
+		user << "<span class='notice'>You pry out the rods, destroying the filter.</span>"
+		qdel(src)
+	if(W.is_open_container())
+		if(!W.reagents.total_volume)
+			W.loc = src
+			cans += W
+			user.drop_item(W, src)
+			user << "<span class='notice'>You add a passive container. It now contains [cans.len].</span>"
+		else
+			if(!beaker)
+				user << "<span class='notice'>You insert an active container.</span>"
+				src.beaker =  W
+				user.drop_item(W, src)
+			else
+				user << "<span class='warning'>There is already an active container.</span>"
+		return
+	else
+		..()
+
+/obj/structure/centrifuge/attack_hand(mob/user as mob)
+	if(cans.len || beaker)
+		for(var/obj/item/O in cans)
+			O.loc = src.loc
+			cans -= O
+		if(beaker)
+			beaker.loc = src.loc
+			beaker = null
+		user << "<span class='notice'>You remove everything from the centrifuge.</span>"
+	else
+		user << "<span class='warning'>There is nothing to eject!</span>"
+
+/obj/structure/centrifuge/verb/flush()
+	set name = "Flush"
+	set category = "Object"
+	set src in view(1)
+	usr << "<span class='notice'>The \the [src] groans as it spits out containers.</span>"
+	while(cans.len>0 && beaker.reagents.reagent_list.len>0)
+		var/obj/item/weapon/reagent_containers/C = cans[1]
+		var/datum/reagent/R = beaker.reagents.reagent_list[1]
+		beaker.reagents.trans_id_to(C,R.id,50)
+		C.loc = src.loc
+		cans -= C
+	if(!cans.len&&beaker.reagents.reagent_list.len)
+		usr << "<span class='warning'>With no remaining containers, the rest of the concoction swirls down the drain...</span>"
+		beaker.reagents.clear_reagents()
+	if(!beaker.reagents.reagent_list.len)
+		usr << "<span class='notice'>The now-empty active container plops out.</span>"
+		beaker.loc = src.loc
+		beaker = null
+		return
+
+/obj/item/weapon/reagent_containers/mortar
+	name = "mortar"
+	desc = "This is a reinforced bowl, used for crushing reagents. Ooga booga Rockstop."
+	icon = 'icons/obj/food.dmi'
+	icon_state = "mortar"
+	flags = FPRINT  | OPENCONTAINER
+	volume = 50
+	amount_per_transfer_from_this = 5
+	//We want the all-in-one grinder audience
+
+	var/list/blend_items = list (

```

Comment:
1. The formatting of the variables is different.
2. Plastellium isn't listed, as it was redundant.
3. The generic food snack option is formatted differently.
4. All of the values for special grinding results are directly in the code.

The grinder list flat-out won't work with my code, and visa versa.